astropy/nddata/decorators.py

```diff
@@ -169,25 +169,23 @@ def support_nddata_decorator(func):
                 else:
                     raise
 
-        # First argument should be data
-        if not func_args or func_args[0] != attr_arg_map.get("data", "data"):
+        data_arg_name = attr_arg_map.get("data", "data")
+        if data_arg_name not in sig.keys():
```

Comment:
```suggestion
        if (data_arg_name := attr_arg_map.get("data", "data")) not in sig.keys():
```

Reply:
I am against doing this because black comes for it because the line it too long and makes it look significantly shitter imo.

Reply:
fair enough